src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/StringConverter.cs

```diff
@@ -15,5 +17,25 @@ public override void Write(Utf8JsonWriter writer, string? value, JsonSerializerO
         {
             writer.WriteStringValue(value);
         }
+
+        internal override string ReadWithQuotes(ReadOnlySpan<byte> utf8Bytes)
+            => JsonReaderHelper.TranscodeHelper(utf8Bytes);
+
+        internal override void WriteWithQuotes(Utf8JsonWriter writer, [DisallowNull] string? value, JsonSerializerOptions options, ref WriteStack state)
```

Comment:
> [DisallowNull] string? value

The reader makes sure this isn't null upstream?

Reply:
Yes, it cannot be null, otherwise the reader would throw since a JSON property name can only be string. The `DisallowNull` attribute is only placed to make this method match with the base signature.

Reply:
In this case `T` is `string?` and forces that the override uses `string?` as well, that's why I was adding the attribute `DisallowNull`.

An alternative to avoid using the attribute would be to change `StringConverter`'s base type, from `JsonConverter<string?>` to `JsonConverter<string>`.